### ISSUE TITLE
Centos 7 Dependency problem on upgrade to icingaweb2 2.5

### ISSUE THREAD

**Issue by vdveldet** [Icinga/icinga-packaging#73]
I have a dependency issue when updating to the latest verion 2.5

[root@suzuki ~]# yum install epel-release
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.cu.be
 * epel: ftp.nluug.nl
 * extras: centos.cu.be
 * updates: centos.cu.be
Package epel-release-7-11.noarch already installed and latest version
Nothing to do

[root@suzuki ~]# yum install http://packages.icinga.com/epel/icinga-rpm-release-7-latest.noarch.rpm
Loaded plugins: fastestmirror
icinga-rpm-release-7-latest.noarch.rpm                                                                                                                             | 4.3 kB  00:00:00     
Examining /var/tmp/yum-root-vnwnYd/icinga-rpm-release-7-latest.noarch.rpm: icinga-rpm-release-7-2.el7.centos.noarch
/var/tmp/yum-root-vnwnYd/icinga-rpm-release-7-latest.noarch.rpm: does not update installed package.
Error: Nothing to do

[root@suzuki ~]# yum update icingaweb2
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.cu.be
 * epel: ftp.nluug.nl
 * extras: centos.cu.be
 * updates: centos.cu.be
Resolving Dependencies
--> Running transaction check
---> Package icingaweb2.noarch 0:2.4.2-1.el7.icinga will be updated
---> Package icingaweb2.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: icingaweb2-vendor-Parsedown = 2.5.0-1.el7.icinga for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: icingaweb2-common = 2.5.0-1.el7.icinga for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: icingacli = 2.5.0-1.el7.icinga for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: php-Icinga = 2.5.0-1.el7.icinga for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: icingaweb2-vendor-dompdf = 2.5.0-1.el7.icinga for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: icingaweb2-vendor-lessphp = 2.5.0-1.el7.icinga for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: icingaweb2-vendor-HTMLPurifier = 1:2.5.0-1.el7.icinga for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: icingaweb2-vendor-JShrink = 2.5.0-1.el7.icinga for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-fpm >= 5.6.0 for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-pgsql for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-mysqlnd for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-ldap for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Running transaction check
---> Package icingacli.noarch 0:2.5.0-1.el7.icinga will be installed
--> Processing Dependency: rh-php71-php-cli >= 5.6.0 for package: icingacli-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: bash-completion for package: icingacli-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: /opt/rh/rh-php71/root/usr/bin/php for package: icingacli-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-fpm >= 5.6.0 for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-pgsql for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-mysqlnd for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-ldap for package: icingaweb2-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-common.noarch 0:2.4.2-1.el7.icinga will be updated
---> Package icingaweb2-common.noarch 0:2.5.0-1.el7.icinga will be an update
---> Package icingaweb2-vendor-HTMLPurifier.noarch 1:2.4.2-1.el7.icinga will be updated
---> Package icingaweb2-vendor-HTMLPurifier.noarch 1:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: 1:icingaweb2-vendor-HTMLPurifier-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-vendor-JShrink.noarch 0:2.4.2-1.el7.icinga will be updated
---> Package icingaweb2-vendor-JShrink.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-vendor-JShrink-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-vendor-Parsedown.noarch 0:2.4.2-1.el7.icinga will be updated
---> Package icingaweb2-vendor-Parsedown.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-vendor-Parsedown-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-vendor-dompdf.noarch 0:2.4.2-1.el7.icinga will be updated
---> Package icingaweb2-vendor-dompdf.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-vendor-dompdf-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-vendor-lessphp.noarch 0:2.4.2-1.el7.icinga will be updated
---> Package icingaweb2-vendor-lessphp.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-vendor-lessphp-2.5.0-1.el7.icinga.noarch
---> Package php-Icinga.noarch 0:2.4.2-1.el7.icinga will be updated
---> Package php-Icinga.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: icingaweb2-vendor-zf1 = 2.5.0-1.el7.icinga for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-intl for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-xml for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-gd for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-pdo for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-mbstring for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Running transaction check
---> Package bash-completion.noarch 1:2.1-6.el7 will be installed
---> Package icingacli.noarch 0:2.5.0-1.el7.icinga will be installed
--> Processing Dependency: rh-php71-php-cli >= 5.6.0 for package: icingacli-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: /opt/rh/rh-php71/root/usr/bin/php for package: icingacli-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-fpm >= 5.6.0 for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-pgsql for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-mysqlnd for package: icingaweb2-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-ldap for package: icingaweb2-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-vendor-HTMLPurifier.noarch 1:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: 1:icingaweb2-vendor-HTMLPurifier-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-vendor-JShrink.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-vendor-JShrink-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-vendor-Parsedown.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-vendor-Parsedown-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-vendor-dompdf.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-vendor-dompdf-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-vendor-lessphp.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-vendor-lessphp-2.5.0-1.el7.icinga.noarch
---> Package icingaweb2-vendor-zf1.noarch 0:2.4.2-1.el7.icinga will be updated
---> Package icingaweb2-vendor-zf1.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: icingaweb2-vendor-zf1-2.5.0-1.el7.icinga.noarch
---> Package php-Icinga.noarch 0:2.5.0-1.el7.icinga will be an update
--> Processing Dependency: rh-php71-php-common >= 5.6.0 for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-intl for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-xml for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-gd for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-pdo for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: rh-php71-php-mbstring for package: php-Icinga-2.5.0-1.el7.icinga.noarch
--> Processing Dependency: /opt/rh/rh-php71/root/usr/bin/php for package: icingacli-2.5.0-1.el7.icinga.noarch
--> Finished Dependency Resolution
Error: Package: icingaweb2-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-mysqlnd
Error: Package: php-Icinga-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-gd
Error: Package: php-Icinga-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-pdo
Error: Package: icingaweb2-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-common >= 5.6.0
Error: Package: icingaweb2-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-pgsql
Error: Package: icingaweb2-vendor-lessphp-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-common >= 5.6.0
Error: Package: icingaweb2-vendor-Parsedown-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-common >= 5.6.0
Error: Package: icingaweb2-vendor-JShrink-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-common >= 5.6.0
Error: Package: php-Icinga-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-intl
Error: Package: icingacli-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-cli >= 5.6.0
Error: Package: icingaweb2-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-fpm >= 5.6.0
Error: Package: icingaweb2-vendor-dompdf-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-common >= 5.6.0
Error: Package: icingaweb2-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-ldap
Error: Package: icingaweb2-vendor-zf1-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-common >= 5.6.0
Error: Package: php-Icinga-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-xml
Error: Package: php-Icinga-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-common >= 5.6.0
Error: Package: 1:icingaweb2-vendor-HTMLPurifier-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-common >= 5.6.0
Error: Package: icingacli-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: /opt/rh/rh-php71/root/usr/bin/php
Error: Package: php-Icinga-2.5.0-1.el7.icinga.noarch (icinga-stable-release)
           Requires: rh-php71-php-mbstring
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

[root@suzuki ~]#

**Comment by dnsmichi** [comment]
https://www.icinga.com/docs/icingaweb2/latest/doc/80-Upgrading/#upgrading-to-icinga-web-2-25x

**Comment by vdveldet** [comment]
I must have overlooked.

```
yum install centos-release-scl
yum install icingaweb2 icingacli
```
Solved the issue of installation.

I still have a minor issue with "Backend icinga is not running", but will take a look and open an new issue if needed.
Thanks for the Help!!
